org-roam.el

```diff
@@ -200,6 +200,26 @@ The search terminates when the first property is encountered."
         (push (cons prop p) res)))
     res))
 
+(defun org-roam--expand-links (content path)
+  "Crawl CONTENT for relative links and expand them.
+PATH should be the root from which to compute the relativity."
+  (let ((dir (file-name-directory path)))
+    (with-temp-buffer
+      (insert content)
+      (goto-char (point-min))
+      ;; Loop over links
+      (while (re-search-forward org-link-bracket-re (point-max) t)
+        (goto-char (match-beginning 1))
+        ;; Strip 'file:'
+        (setq link (substring (match-string 1) 5))
+        ;; Delete relative link
+        (when (f-relative-p link)
+          (delete-region (match-beginning 1)
+                         (match-end 1))
+          (insert (expand-file-name
+                   (concat dir link)))))
+      (buffer-string))))
```

Comment:
This needs to be a bit more robust, since it will break all non `file:` based links.

I've also tried to include an absolute `file:` link to somewhere outside of `org-roam-directory` and it breaks it too. Not sure though if it someone would do so in a real case?

Reply:
> This needs to be a bit more robust, since it will break all non `file:` based links.

Thanks for spotting this.  I'd shielded the code at the start, but through the iterations, I've forgot to reimplement the check.

> I've also tried to include an absolute `file:` link to somewhere outside of `org-roam-directory` and it breaks it too. Not sure though if someone would do so in a real case?

What do you mean by breaking?  It's working for me.
![screenshot-E0cumC3rgA](https://user-images.githubusercontent.com/12202828/80293134-e8c22500-875c-11ea-9b79-b9b69a23f9ee.png)

Reply:
> What do you mean by breaking?

Huh, It didn't follow it from the org-roam-buffer and instead it would open for me `foo1.org` file. Cannot reproduce this after restarting Emacs.

Reply:
I think this is expected behaviour.  Clicking anywhere on the text which isn't an `org-roam-link` or `org-roam-link-current` brings you to the file where the link was found.

Reply:
Now I can reproduce and yes, you're correct. This logic is written in `org-roam-open-at-point`.

Reply:
> This needs to be a bit more robust, since it will break all non `file:` based links.

Fixed in 80f6a8affc2788d744534b11e4cc79d45e330b11.